applications/luci-app-ddns/root/usr/libexec/rpcd/ddns

```diff
@@ -0,0 +1,328 @@
+#!/usr/bin/env lua
+
+local json = require "luci.jsonc"
+local nixio = require "nixio"
+local fs   = require "nixio.fs"
+local UCI = require "luci.model.uci"
+local sys  = require "luci.sys"
+local util = require "luci.util"
+
+local luci_helper = "/usr/lib/ddns/dynamic_dns_lucihelper.sh"
+local srv_name    = "ddns-scripts"
+
+-- convert epoch date to given format
+local function epoch2date(epoch, format)
+	if not format or #format < 2 then
+		local uci = UCI.cursor()
+		format    = uci:get("ddns", "global", "ddns_dateformat") or "%F %R"
+		uci:unload("ddns")
+	end
+	format = format:gsub("%%n", "<br />")	-- replace newline
+	format = format:gsub("%%t", "    ")	-- replace tab
+	return os.date(format, epoch)
+end
+
+-- function to calculate seconds from given interval and unit
+local function calc_seconds(interval, unit)
+	if not tonumber(interval) then
+		return nil
+	elseif unit == "days" then
+		return (tonumber(interval) * 86400)	-- 60 sec * 60 min * 24 h
+	elseif unit == "hours" then
+		return (tonumber(interval) * 3600)	-- 60 sec * 60 min
+	elseif unit == "minutes" then
+		return (tonumber(interval) * 60)	-- 60 sec
+	elseif unit == "seconds" then
+		return tonumber(interval)
+	else
+		return nil
+	end
+end
+
+local methods = {
+	get_services_log = {
+		args = { service_name = "service_name" },
+		call = function(args)
+			local result = "File not found or empty"
+			local uci = UCI.cursor()
+
+			local dirlog = uci:get('ddns', 'global', 'ddns_logdir') or "/var/log/ddns"
+
+			if args.service_name and fs.access("%s/%s.log" % { dirlog, args.service_name }) then
```

Comment:
Note that this could theoretically allow for path traversal attacks, e.g. if "service_name" is set to something like `../../../some/different/path`.

Apart from that; do we actually need this call? We could use the normal [`fs.read()`](http://openwrt.github.io/luci/jsapi/LuCI.fs.html#read) procedure and figure out the `ddns_logdir` value on the client side.

Reply:
problem is that fs.read needs to have permission to access the file in acl.d 

I moved this here to bypass this restriction... Any idea on how to solve this?

Reply:
Ah I understand... in this case it makes sense to have this extra procedure.

Reply:
To prevent path traversal i can add a validate function to make the string "../" not allowed